sdk/src/reader.rs

```diff
@@ -0,0 +1,244 @@
+// Copyright 2024 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+//! The Reader provides a way to read a manifest store from an asset.
+//! It also performs validation on the manifest store.
+
+#[cfg(feature = "file_io")]
+use std::fs::{read, File};
+use std::io::{Read, Seek, Write};
+
+use async_generic::async_generic;
+
+#[cfg(feature = "file_io")]
+use crate::error::Error;
+use crate::{
+    claim::ClaimAssetData, error::Result, manifest_store::ManifestStore,
+    settings::get_settings_value, status_tracker::DetailedStatusTracker, store::Store,
+    validation_status::ValidationStatus, Manifest,
+};
+
+/// A reader for the manifest store.
+#[derive(Debug)]
+pub struct Reader {
+    pub(crate) manifest_store: ManifestStore,
+}
+
+impl Reader {
+    /// Create a manifest store Reader from a stream.
+    /// # Arguments
+    /// * `format` - The format of the stream.
+    /// * `stream` - The stream to read from.
+    /// # Returns
+    /// A reader for the manifest store.
+    /// # Errors
+    /// If the stream is not a valid manifest store.
+    /// validation status should be checked for non severe errors
+    /// # Example
+    /// ```no_run
+    /// use std::io::Cursor;
+    ///
+    /// use c2pa::Reader;
+    /// let mut stream = Cursor::new(include_bytes!("../tests/fixtures/CA.jpg"));
+    /// let reader = Reader::from_stream("image/jpeg", stream).unwrap();
+    /// println!("{}", reader.json());
+    /// ```
+    #[async_generic(async_signature(
```

Comment:
I don't think you need `async_signature` here since the sync and async signatures are the same.l